platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java

```diff
@@ -582,42 +582,28 @@ public void removeSnapshots(String owner, String workspaceId, ProjectBinding pro
      */
     public InstanceProcess exec(final String machineId, final Command command, String clientOutputChannel)
             throws NotFoundException, MachineException {
+        return exec(machineId, command, clientOutputChannel, null, true);
+    }
+
+    public InstanceProcess exec(final String machineId,
+                                final Command command,
+                                final String clientOutputChannel,
```

Comment:
why this method contains both processLogger and clientOutputChannel?
I propose separate it into 2 methods. First accepts only process logger. Another one accepts only client output channel, creates process logger and calls the first one.
Wdyt?